### ISSUE TITLE
Language/multiple layers support for data fetching

### ISSUE THREAD

**Issue by pfrang** [enonic/nextjs-adapter#90]
Working with multiple layers, there are different endpoints. Currently CONTENT_API_DRAFT/MASTER seems to be the only source data is fetched from in the adapter. Maybe a language header can be added so that the layer can be chosen in XP, graphql.js.

**Comment by pfrang** [comment]
Yes, exactly. It is basically to just fetch/update english content when on the english layer, which lives in a different endpoint. 
After a good extensive meeting with Pavel earlier today, he suggested to post the issues we went through together here. 

If Guillotine API updates so that all projects/layers live on a single endpoint, that would solve this issue. When is that planned to be released? Maybe Pavel should chime in about this, because what we concluded on was on the short term to add a language/locale parameter in the header from the adapter, so we can specify layer on the xp /graphql side based on that.

**Comment by sigdestad** [comment]
Hi Peder.

The current SDK has a setup using exactly one layer per Next.js instance. I'm not sure if I understand your use-case correctly? Maybe you could provide some more deails on what you want to do?

**Comment by pfrang** [comment]
Hi Thomas,

Sorry for the late reply, totally forgot. 
Me and Pavel went through it in detail in a call with shared screen etc. he is aware and can probably explain better than me.

When the next application sends the request to Enonic to retrieve the data, we either hit the english layer or the norwegian layer based on the current "locale" in the next application. We need to distinguish on the enonic/graphql whether the requested should be queried on the english or nowegian layer. Basically our internal "guillotine-api" works like this (picture), so if we can manage this somehow on the graphql side, parsing the incoming the request with some sort of language header i.e, we can "reroute" the request to fetch the corresponding layer.

![image](https://user-images.githubusercontent.com/85552255/225360651-aeee5105-b312-45e7-8b74-58a6f46ff768.png)

**Comment by sigdestad** [comment]
Hi Peder.

Still have a problem understanding the use-case. If you are using our Next.js SDK - it is designed to communicate with one specific API endpoint - which is configured at compile time? As such, you will know both compile and runtime which "language" the front-end is running.

So, how and where does a header from us come into play?

**Comment by pfrang** [comment]
Im not a Next expert, but currently on build time, we build the entire application for both languages, due to the setup in next.config.js with the i18n config locales set to "no" and "en" with our internal guillotine api setup where we change the enonic endpoint based on locales, in the code snippet above.

A header comes into play, because on the enonic side, in graphql.js, we can live with only hitting one specific endpoint (norwegian layer) IF we can reroute the request and change the layer based on some locale header.
